tools/packbeam/packbeam.c

```diff
@@ -148,328 +194,380 @@ static void *pack_beam_fun(void *accum, const void *section_ptr, uint32_t sectio
         return NULL;
     }
 
-    FILE *pack = (FILE *)accum;
+    FILE *pack = (FILE *) accum;
     size_t r = fwrite(section_ptr, sizeof(unsigned char), section_size, pack);
     if (r != section_size) {
         return NULL;
     }
     return accum;
 }
 
-FileData read_file_data(FILE *file)
+static bool validate_pack_files(char *output_file, char **input_files, size_t files_n)
 {
-    fseek(file, 0, SEEK_END);
-    size_t size = ftell(file);
-    fseek(file, 0, SEEK_SET);
-    uint8_t *data = malloc(size);
-    if (!data) {
-        fprintf(stderr, "Unable to allocate %zu bytes\n", size);
-        exit(EXIT_FAILURE);
-    }
-    assert_fread(data, size, file);
+    FileData file_data = (FileData){ .data = NULL, .size = 0 };
 
-    FileData file_data = {
-        .data = data,
-        .size = size
-    };
-    return file_data;
-}
-
-static bool is_avm_file(FILE *file)
-{
-    FileData file_data = read_file_data(file);
-    bool ret = avmpack_is_valid(file_data.data, file_data.size);
-    free(file_data.data);
-    return ret;
-}
+    TRY(safe_read_file(output_file, &file_data));
```

Comment:
This looks like a problem. Trying to read and verify that the not yet created avm archive is valid... this check should be done after the files have been packed into the output archive.

Reply:
I can successfully create an archive with lines 209-214 commented out...But AtomVM is unable to run the hello_world.avm with an error:
```
Unable to locate entrypoint.
```